package.json

```diff
@@ -16,18 +16,29 @@
   },
   "sideEffects": false,
   "type": "module",
-  "exports": "./index.cjs",
+  "main": "./index.cjs",
+  "exports": {
+    ".": {
+      "types": "./index.d.cts",
+      "require": "./index.cjs",
+      "default": "./index.cjs"
```

Comment:
```suggestion
      "types": "./index.d.cts",
      "default": "./index.cjs"
```

Reply:
I'm not 100% certain about this

Not sure all runtimes and bundlers fall back to `"default"` when `"require"` (or `"import"`) is missing

Node.js resolution works of course:

```shell
node --eval "console.log(require.resolve('@prettier/sync'))" --conditions import
node --eval "console.log(require.resolve('@prettier/sync'))" --conditions require
```

Tools that validate **package.json** can also reach it via the new `"./*": "./*"` export too:

```shell
node --eval "console.log(require.resolve('@prettier/sync/package.json'))" --conditions import
node --eval "console.log(require.resolve('@prettier/sync/package.json'))" --conditions require
```

Reply:
The type definitions are co-located with the JavaScript file. (`index.cjs` / `index.d.cts`). This means the `types` field isn’t needed. The entire `exports` can be reverted.

Also additional unrelated exports have been added.

Reply:
Sounds good, I wasn't sure if every type aware bundler or runtime would still find the types 👍

Removing `"exports"` entirely will prevent `./package.json` and `./index.cjs` being resolved directly, which can cause issues with linters that validate package exports, or TypeScript [`{ "resolvePackageJsonExports": true }`](https://www.typescriptlang.org/tsconfig#resolvePackageJsonExports)

I've split out separate commits for package resolution issues

Happy to move them to separate PR if hand-written types are preferred